### ISSUE TITLE
[Feature request] block by instance and keywords

### ISSUE THREAD

**Issue by donnielrt** [sheodox/alexandrite#71]
It would be awesome to be able to block posts by keywords in settings. Additionally, after [3869](https://github.com/LemmyNet/lemmy/pull/3869) merges, it would be amazing to be able to block instances from the feed as well.

Thanks!

**Comment by sheodox** [comment]
Oo nice, didn't know that PR was coming. Yes of course I will add that as soon as I can! :) 

I've been trying to avoid filtering posts out on the client-side in Alexandrite because the less content I can show you the more content I have to ask the server for, but I might come up with a middle-ground to allow me to add extra stuff like this by just showing a smaller "blocked post from xyz" in the feed. Unsure exactly what I'd do but I'm still thinking about it. Having that supported on Lemmy's backend is the more complete solution so I'm excited for that PR.

**Comment by donnielrt** [comment]
Yeah, I'm really excited for backend support for that feature instead of clients having to block individually as well! 

I suppose that still leaves keyword blocking. Most other clients seem to already have added that as a client-side feature, but I agree that it doesn't seem right to have it per-client instead of shared via the lemmy backend. Looks like there's a [feature request open for it](https://github.com/LemmyNet/lemmy/issues/3710), so might make sense to wait till the lemmy devs weigh in on that.

**Comment by sheodox** [comment]
Oops so the "as soon as I can" was a lie in retrospect. 🙃

But instance blocks (and a completely redesigned "Blocks" settings tab) is coming in #114! 

It was a fairly big change to redesign that tab, so if you're still interested and wanted to help test you could check the "Preview URL" on the cloudflare bot comment on that PR. Keep in mind as I make more changes to the PR that url changes, so if you wanted to use the preview version while it's in development you'd want to check the comment periodically.

The old Blocks tab was probably the most basic and neglected page on Alexandrite, but now it actually looks good, and adds search functionality for communities/users/instances similar to lemmy-ui.

I still agree with my statement about the keyword block thing. If it's supported by lemmy I'd for sure do it, there was some activity on feature request ticket you linked a month ago so maybe it'd be good to wait and see if something comes of it. 

Here's what the new page looks like:
![Image](https://github.com/user-attachments/assets/3020588f-7867-4084-a2cc-19bc9862487c)
